artifacts-wg/charter/charter.md

```diff
@@ -0,0 +1,49 @@
+# Purpose
+In the distributed world of cloud-native computing different artifacts and packages are used to transport configuration and code for the many services and capabilities that comprise and support workloads and applications. As an example, today ArtifactHub advertises that it can crawl about 20 such artifact types [1]. These "cloud-native" bundles add to the previous proliferation of package and artifact types for software dependencies such as Maven and npm and for system packages like deb and rpm.
+
+This abundance of package types and formats adds complexity and risk for cloud application developers and architects trying to provision and run cloud-native workloads. Different tools and controllers must be carefully learned and installed to bundle, unbundle, verify and deploy each artifact type. This complexity slows product development by users, impedes innovation and collaboration amongst projects, and increases risks of insecure configuration.
+
+To reduce this complexity and facilitate collaboration and innovation, WG Artifacts will gather stakeholders from many CNCF and open source projects offering packaging, distribution and deployment mechanisms for bundles of configuration and code.
+
+**The WG will:**  
+* **Develop common patterns**  
+*  **Advocate for innovative projects**  
+*  **Gather end user feedback**
+**for packaging, discovery, distribution, and deployment mechanisms for** **cloud-native software artifacts.**
+
+# Alignment with TAG App Delivery and CNCF
+TAG App Delivery's charter [2] calls on the TAG to develop best practices, foster collaboration between projects, improve interoperability and propose new initiatives and projects related to the following domains which this working group would target.
+
+* Application bundling and deployment
+* Package management
+* Application delivery workflow and strategy
+* Configuration source-driven workflow
+* Release management
+
+The work on extensions, patterns and prototypes in this CNCF TAG complements work on official specifications in the Open Containers Initiative (OCI) itself.
+
+# Goals
+* Gather existing schemas used via OCI Artifacts as described in https://github.com/opencontainers/artifacts
+* Develop a pattern for dynamically discovering existing Artifact schemas
+This in contrast to asking all schema developers to publish media types via IANA.
```

Comment:
Iana registration feels like a specific thing that might be part of the discussion, but is it a goal to remove?
Iana registration was used to solve a clearing house of ownership for a specific `artifactType`, It solves the conflicting file extensions type problem. What keeps someone from defining different versions of "helm", or "spdx"? 
See [Defining a Unique Artifact Type](https://github.com/opencontainers/artifacts/blob/main/artifact-authors.md#defining-a-unique-artifact-type)
To avoid bike shedding specifics here, perhaps we can remove line 28, and explore the dynamic discovery?

Reply:
@SteveLasker, I agree that we should not call out IANA here, however I do think that defining an extensibility model is important for the working group to have as a goal. Ideally, it would take administrative burden off of registry operators and publishers by dynamically registering schemas as they are used to describe artifacts and their relationships.

Reply:
In an effort to not over promise we might want to make a section called "Stretch Goals" to cover topics including decentralization and federation.  We could also include as a stretch goal something about how OCI could be a package indices replacement for PyPI, Apt, etc.

Reply:
I think the "package index" idea is covered under the "Search Query API" goal.  It might make sense to add "package index" to the "Activities" section where we talk about npm, PyPI, etc.  We only discuss the artifacts in that activity (i.e., how we might store them) and not how we might search for them (i.e., the package index functionality).

Reply:
To be frank I expect we'll keep static registration but introduce experiments for dynamic type/schema discovery alongside that.

I think we've removed this topic from the charter in any case, LGTM.